src/contract.rs

```diff
@@ -142,6 +143,29 @@ impl<T: Transport> Instance<T> {
     {
         Ok(self.method(name, params)?.view())
     }
+
+    /// Returns a log stream for the specified event that emits a new log for
+    /// every new event emitted after the stream was created.
+    pub fn event<S>(&self, name: S) -> AbiResult<LogStream<T>>
+    where
+        S: AsRef<str>,
+    {
+        let event = self.abi.event(name.as_ref())?;
+        if event.anonymous {
+            todo!("anonymous events are currently not supported");
+        }
+
+        let filter = FilterBuilder::default()
+            .address(vec![self.address])
+            .topics(Some(vec![event.signature()]), None, None, None)
+            .build();
+
+        Ok(LogStream::new(
+            self.web3(),
+            filter,
+            std::time::Duration::from_secs(7),
```

Comment:
That seems like a pretty arbitrary and slow polling duration. maybe 1s is better?